src/ert/config/analysis_module.py

```diff
@@ -33,6 +34,23 @@ class VariableInfo(TypedDict):
 DEFAULT_IES_DEC_STEPLENGTH = 2.50
 DEFAULT_ENKF_TRUNCATION = 0.98
 DEFAULT_IES_INVERSION = 0
+DEFAULT_LOCALIZATION = False
+# Default threshold is a function of ensemble size which is not available here.
+DEFAULT_LOCALIZATION_CORRELATION_THRESHOLD = -1
+
+
+def correlation_threshold(ensemble_size: int, user_defined_threshold: float) -> float:
+    """Decides whether or not to use user-defined or default threshold.
+
+    Default threshold taken from luo2022,
+    Continuous Hyper-parameter OPtimization (CHOP) in an ensemble Kalman filter
+    Section 2.3 - Localization in the CHOP problem
+    """
+    default_threshold = 3 / math.sqrt(ensemble_size)
+    if user_defined_threshold == -1:
```

Comment:
Please don't compare floats using `==`.

------------

In fact, I think using `None` is better here. More typical Python, more readable:

```python
if user_defined_threshold is None:
    return default_threshold
```

------------

Come to think of it, I believe the (1) decision to use the default or user defined threshold and (2) retrieving the default threshold are two separate concerns. I think this is better:

```python
threshold = # either a float in [0, 1], or None
self.threshold = correlation_threshold(ensemble_size) if threshold is None else threshold
```
In other words, remove `user_defined_threshold` entirely. The decision is already taken before the function is called (by setting it to `-1` / `None`), so no need to propagate the data down into the function.

Reply:
I agree with this.
Could you make an issue with this?